### PR TITLE
Feat/#tc 66 create a modal for creating discounts

### PR DESCRIPTION
**Pull request by Teodor25**
### This pull request closes:

🏁 Jira Issue: [TC-66](https://treecreate.atlassian.net/browse/TC-66)

### Which service(s) does this issue affect:

- [ ] 🛰️ API
- [ ] 🛒 webstore
- [x] 🎓 admin-page

### Types of changes

- [ ] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [x] 🧱 New feature (a non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🤖 CI/CD (a change to the CI/CD pipeline)
- [ ] ⚗️ Refactoring (an update to some already existing code)

### Documentation Updates

- [ ] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [x] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [ ] Tested on Safari
- [x] Tested on Chrome
- [ ] Tested on Mozilla Firefox

### What changes have been done within this issue?

Created a dialog for creating a discount. 

### How should this be manually tested?

When in discounts page, click "create" and the dialog will open. Fill the form and press create discount and the discount will be created.

### CODE REVIEW

File: apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        startsAt: this.getDateTime(this.createDiscountForm.get('startsAt')?.value) || new Date(),
+        isEnabled: this.createDiscountForm.get('isEnabled')?.value,
+        remainingUses: this.createDiscountForm.get('remainingUses')?.value,
+        totalUses: 0,
+        amount: this.createDiscountForm.get('amount')?.value,
+        type: this.createDiscountForm.get('type')?.value,
+      })
+      .subscribe(
+        (data: IDiscount) => {
+          console.log(data);
```
Comment (Kwandes) [inline apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts]: You should remove the console log (don't spam the console)
```suggestion
```

File: apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          this.snackbar.open('Discount created!', 'HOLY SH***', { duration: 2500 });
+          this.dialog.closeAll();
```
Comment (Kwandes) [inline apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts:66]: You should update the discount list / re-fetch all discounts to display the newly created discount

File: apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   *
+   * @param discountService
+   * @param snackbar
+   * @param dialog
+   */
+  constructor(private discountService: DiscountsService, private snackbar: MatSnackBar, private dialog: MatDialog) {
+    this.createDiscountForm = new FormGroup({
+      discountCode: new FormControl('', [Validators.maxLength(50), Validators.minLength(3), Validators.required]),
+      startsAt: new FormControl(''),
+      expiresAt: new FormControl('', [Validators.required]),
```
Comment (Kwandes) [inline apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts:30]: Expires at is not required and can be null in the discount

File: apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        </button>
+        <button
+            mat-flat-button
+            color="primary"
+            class="action-button"
+            data-cy="discount-create-btn"
+            type="submit"
+            [disabled]="createDiscountForm.invalid"
+        >
+            Create
```
Comment (Kwandes) [inline apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.html]: (Optional) Should have a tooltip explaining why the discount is disabled
(Optional) The button should be disabled while isLoading (the request is being made)

File: apps/admin-page/src/app/pages/discounts/discounts.component.html [hunk 5]
```diff
@@ -2,7 +2,13 @@
     <ng-template #content>
         <div class="header">
             <div matTooltip="Not implemented yet.">
```
Comment (Kwandes) [inline apps/admin-page/src/app/pages/discounts/discounts.component.html]: Remove the tooltip for the button not being implemented

File: apps/admin-page-e2e/src/integration/pages/discounts.spec.ts [hunk 6]
```diff
@@ -89,3 +89,42 @@ describe('discountsPage', () => {
     cy.get('[data-cy=discounts-enable-btn]').should('have.length', mockDiscounts.length);
   });
 });
+
+describe('create discount dialog', () => {
```
Comment (Kwandes) [inline apps/admin-page-e2e/src/integration/pages/discounts.spec.ts:96]: You're missing a test for failure scenario (creation request fails)
Should display a snackbar, the buttons should remain enabled and the dialog should stay open

**Review by Kwandes — CHANGES_REQUESTED** [review]
I have a few mew minor tweaks and a few optional improvements.
Good job on the code, and thanks for taking care of this issue for me

File: apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts (hunk 3 above)
Comment (Teodor25) [reply apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts:30]: I want it to be required so an end date is set.

**Review by Kwandes — APPROVED** [review]

**Pull request merged**